net/snowflake/Makefile

```diff
@@ -0,0 +1,144 @@
+include ../../lang/golang/golang-package.mk
+
+define Package/snowflake/Default
+  TITLE:=Snowflake
+  URL:=https://snowflake.torproject.org/
+  DEPENDS:=$(GO_ARCH_DEPENDS)
+  SECTION:=net
+  CATEGORY:=Network
+  USERID:=snowflake:snowflake
+endef
+
+define Package/snowflake-broker
+$(call Package/snowflake/Default)
+  TITLE+= Broker
+endef
+
+define Package/snowflake-client
+$(call Package/snowflake/Default)
+  TITLE+= Client
+endef
+
+define Package/snowflake-distinctcounter
+$(call Package/snowflake/Default)
+  TITLE+= Distinct Counter
+endef
+
+define Package/snowflake-probetest
+$(call Package/snowflake/Default)
+  TITLE+= Probe test
+endef
+
+define Package/snowflake-proxy
+$(call Package/snowflake/Default)
+  TITLE+= Proxy
+endef
+
+define Package/snowflake-server
+$(call Package/snowflake/Default)
+  TITLE+= Server
+endef
+
+define Package/snowflake/description/Default
+Snowflake is a system that allows people from all over the world to
+access censored websites and applications. Similar to how VPNs assist
+users in getting around Internet censorship, Snowflake helps you avoid
+being noticed by Internet censors by making your Internet activity appear
+as though you're using the Internet for a regular video or voice call.
+endef
+
+define Package/snowflake-broker/description
+$(call Package/snowflake/description/Default)
+
+This package provides the Snowflake broker service.
+endef
+
+define Package/snowflake-client/description
+$(call Package/snowflake/description/Default)
+
+This package contains the Snowflake client which provides the bridge to TOR.
+endef
+
+define Package/snowflake-distinctcounter/description
+$(call Package/snowflake/description/Default)
+
+This package provides the Snowflake distinct counter service.
+endef
+
+define Package/snowflake-probetest/description
+$(call Package/snowflake/description/Default)
+
+This package provides the Snowflake probe test.
+endef
+
+define Package/snowflake-proxy/description
+$(call Package/snowflake/description/Default)
+
+This package provides the standalone Snowflake proxy.
+endef
+
+define Package/snowflake-server/description
+$(call Package/snowflake/description/Default)
+
+This package provides the Snowflake server.
+endef
+
+define Package/snowflake-broker/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(GO_PKG_BUILD_BIN_DIR)/broker $(1)/usr/bin/snowflake-broker
+endef
+
+define Package/snowflake-client/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(GO_PKG_BUILD_BIN_DIR)/client $(1)/usr/bin/snowflake-client
+endef
+
+define Package/snowflake-distinctcounter/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(GO_PKG_BUILD_BIN_DIR)/distinctcounter $(1)/usr/bin/snowflake-distinctcounter
+endef
+
+define Package/snowflake-probetest/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(GO_PKG_BUILD_BIN_DIR)/probetest $(1)/usr/bin/snowflake-probetest
+endef
+
+define Package/snowflake-proxy/install
+	$(INSTALL_DIR) $(1)/usr/bin $(1)/etc/init.d
```

Comment:
I don't know why, but all(?) other go packages do something like this:
`$(call GoPackage/Package/Install/Bin,$(PKG_INSTALL_DIR))`
in front of installing the binaries.

Reply:
Because most Go package builds only result in a single package containing everything. In order to split up the components into several packages I choose to not use `GoPackage/Package/Install/Bin`.

Reply:
Than maybe some other packages are wrong? Look at the [Makefile from nebula](https://github.com/openwrt/packages/blob/master/net/nebula/Makefile). They have two modules, nebula and nebula-cert. They copy the nebula binary from [here](https://github.com/openwrt/packages/blob/master/net/nebula/Makefile#L63) and for cert [here](https://github.com/openwrt/packages/blob/master/net/nebula/Makefile#L71). However, they use `$(PKG_INSTALL_DIR)` instead of `$(GO_PKG_BUILD_BIN_DIR)`.

I don't know why they do it. I think your solution is better since it seems unnecessary to install and than copy again the same binary. Or maybe I misinterpret the Makefile code of nebula.

Reply:
Yeah, I've seen that and then looked at golang-package.mk and the scripts in lang/golang/ and concluded that `$(call GoPackage/Package/Install/Bin,$(1))` is a useful thing to do if really everything should end up in one package. But in case things should be split up there is no need for all the moving/copying of files as accessing them directly is just as easy and convenient.

Reply:
Thanks. Resolved.